src/instruments/src/SD/Pages/Fctl/Fctl.tsx

```diff
@@ -1,738 +1,390 @@
-
-export const FctlPage = () => {
-    const [aileronLeftDeflectionState] = useSimVar('AILERON LEFT DEFLECTION PCT', 'percent over 100', 50);
-    const [aileronRightDeflectionState] = useSimVar('AILERON RIGHT DEFLECTION PCT', 'percent over 100', 50);
+import './Fctl.scss';
+import { PageTitle } from '../../Common/PageTitle';
+import { EcamPage } from '../../Common/EcamPage';
+import { SvgGroup } from '../../Common/SvgGroup';
 
-    const [elevatorDeflectionState] = useSimVar('ELEVATOR DEFLECTION PCT', 'percent over 100', 50);
+setIsEcamPage('fctl_page');
 
-    const [elac1SwitchState] = useSimVar('L:A32NX_FBW_ELAC_SWITCH:1', 'boolean', 1000);
-    const [elac1FailState] = useSimVar('L:A32NX_FBW_ELAC_FAILED:1', 'boolean', 1000);
+interface ComponentPositionProps {
+    x: number,
+    y: number,
+}
 
-    const [elac2SwitchState] = useSimVar('L:A32NX_FBW_ELAC_SWITCH:2', 'boolean', 1000);
-    const [elac2FailState] = useSimVar('L:A32NX_FBW_ELAC_FAILED:2', 'boolean', 1000);
+interface ComponentSidePositionProps {
+    side: 'left' | 'right'
+}
 
-    const [sec1SwitchState] = useSimVar('L:A32NX_FBW_SEC_SWITCH:1', 'boolean', 1000);
-    const [sec1FailState] = useSimVar('L:A32NX_FBW_SEC_FAILED:1', 'boolean', 1000);
+interface HydraulicSystemPairProps {
+    leftHydraulicSystem: HydraulicSystem,
+    rightHydraulicSystem: HydraulicSystem,
+}
 
-    const [sec2SwitchState] = useSimVar('L:A32NX_FBW_SEC_SWITCH:2', 'boolean', 1000);
-    const [sec2FailState] = useSimVar('L:A32NX_FBW_SEC_FAILED:2', 'boolean', 1000);
+type HydraulicSystem = 'B' | 'Y' | 'G';
 
-    const [sec3SwitchState] = useSimVar('L:A32NX_FBW_SEC_SWITCH:3', 'boolean', 1000);
-    const [sec3FailState] = useSimVar('L:A32NX_FBW_SEC_FAILED:3', 'boolean', 1000);
+interface HydraulicSystemAvailable {
+    available: boolean
+}
 
-    const [pitchTrimState] = useSimVar('ELEVATOR TRIM INDICATOR', 'Position 16k', 50);
+interface HydraulicsContext {
+    G: HydraulicSystemAvailable;
+    Y: HydraulicSystemAvailable;
+    B: HydraulicSystemAvailable;
+}
+const HydraulicContext = React.createContext<HydraulicsContext>({
+    G: { available: false },
+    Y: { available: false },
+    B: { available: false },
+});
 
-    const [rawPitchTrim, setRawPitchTrim] = useState(0);
-    const [rudderAngle, setRudderAngle] = useState(0);
-    const [rudderTrimAngle, setRudderTrimAngle] = useState(0);
-    const [maxAngleNorm, setMaxAngleNorm] = useState(1);
+export const FctlPage = () => {
+    const [engine1State] = useSimVar('TURB ENG N2:1', 'Percent', 1000);
+    const [engine2State] = useSimVar('TURB ENG N2:2', 'Percent', 1000);
 
-    useEffect(() => {
-        let rPT = pitchTrimState / 1213.6296;
-        // Cap pitch trim at 13.5 up, 4 down
-        if (rPT > 16384.0) {
-            rPT = 16384.0;
-        } else if (rPT < -4854) {
-            rPT = -4854;
-        }
-        setRawPitchTrim(rPT);
-    }, [pitchTrimState]);
+    const hydraulicContext: HydraulicsContext = {
```

Comment:
Now that the hydraulics system has been implemented, it would be good to use the actual SimVars for hydraulic system availability rather than the surrogate I implemented based on N2 state.

Reply:
That will/should come in a separate PR. This PR aims to rewrite the code to React.

Reply:
It is written in React already. You're refactoring it, which is fine and an improvement on my original code. The SimVars and function to check for lowpressure are already in the original code (just commented out, lines 77++).

Reply:
I see. Sorry about that. There are by now three or four PRs related to React. Most of them are from old to React. I finished this PR's code 12 days ago and HYD came in later. Feel free to add the changes in after this PR is completed. I don't want to mix in multiple concerns in one PR.